src/main/java/ml/shifu/shifu/udf/PopulationCounterSaltSumUDF.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright [2013-2015] PayPal Software Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ml.shifu.shifu.udf;
+
+import ml.shifu.guagua.util.MemoryUtils;
+import ml.shifu.shifu.container.obj.ColumnConfig;
+import ml.shifu.shifu.core.ColumnStatsCalculator;
+import ml.shifu.shifu.core.dtrain.wdl.WDLWorker;
+import ml.shifu.shifu.udf.stats.CategoryCounter;
+import ml.shifu.shifu.udf.stats.Counter;
+import ml.shifu.shifu.udf.stats.NumericCounter;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.pig.data.DataBag;
+import org.apache.pig.data.Tuple;
+import org.apache.pig.data.TupleFactory;
+import org.apache.pig.impl.logicalLayer.schema.Schema;
+import org.apache.pig.impl.util.Utils;
+import org.apache.pig.parser.ParserException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Calculate the counter for each bin
+ */
+public class PopulationCounterSaltSumUDF extends AbstractTrainerUDF<Tuple> {
+
+    public static Logger logger = LoggerFactory.getLogger(PopulationCounterSaltSumUDF.class);
+
+    private Counter counter;
+    private int index;
+
+    // DO NOT use this constructor
+    private PopulationCounterSaltSumUDF(String source, String pathModelConfig, String pathColumnConfig)
+        throws IOException {
+        super(source, pathModelConfig, pathColumnConfig);
+        this.index = 1;
+    }
+
+    public PopulationCounterSaltSumUDF(String source, String pathModelConfig, String pathColumnConfig, String index)
+        throws IOException {
+        super(source, pathModelConfig, pathColumnConfig);
+        this.index = Integer.valueOf(index);
```

Comment:
Any exception here, or exception processing from Integer.valueOf is fine?

Reply:
here super throw IOException